src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2335,7 +2335,9 @@ class MergeStatementSegment(BaseSegment):
             ),
         ),
         Dedent,
+        Conditional(Indent, indented_using_on=True),
         Ref("JoinOnConditionSegment"),
+        Conditional(Dedent, indented_using_on=True),
```

Comment:
Hey @tunetheweb could you double check if I'm using the correct configuration here? I had to use a different configuration to what you suggested in https://github.com/sqlfluff/sqlfluff/issues/3091#issuecomment-1098752153

Reply:
Yes that looks correct. Sorry it's slightly confusing for the existing use case, in that it Dedents the defaulted indent, if that is false, rather than indents if true.

Your addition here is correct (and much easier to understand since it doesn't have to dedent an existing dedent!).